department-of-veterans-affairs/caseflow
# Add front end validation to receipt date field

Resolves CASEFLOW-1566

### Description
Add validation to the receipt date field to ensure that the date entered is not prior to the AMA start date

This should be considered a quick fix to prevent any more intakes from being entered with an early receipt date, since we're just using vanilla yup to validate the data.

A good IP ticket for this coming IP sprint would be to refactor the way these forms work, since they're not actually html forms so there's not a real "submission" that we could use the more robust yupResolver to enhance the validations here. The refactor would include replacing the Class based components with functional components which would allow us to use form hooks. 

### Acceptance Criteria
- If a user enters a date prior to the AMA start date, chooses unrecognized claimant, and tries to move to next step, they should receive an error informing them that the receipt date cannot be before the AMA date

### Testing Plan
1. Start an NOD appeal
2. On the review page, enter a receipt date prior to 02/19/2019
3. Choose Yes for Claimant other than someone listed
4. Choose claimant not listed
5. Click continue to next step
6. See that there is an error on the receipt date field


### User Facing Changes

 BEFORE|AFTER
 ---|---

### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*
